patches/telemetry.diff

```diff
@@ -0,0 +1,228 @@
++		throw new Error('unimplemented');
++	}
++
++	public trackEvent(options: appInsights.Contracts.EventTelemetry): void {
++		if (!options.properties) {
++			options.properties = {};
++		}
++		if (!options.measurements) {
++			options.measurements = {};
++		}
++
++		try {
++			const cpus = os.cpus();
++			options.measurements.cores = cpus.length;
++			options.properties['common.cpuModel'] = cpus[0].model;
++		} catch (error) {}
++
++		try {
++			options.measurements.memoryFree = os.freemem();
++			options.measurements.memoryTotal = os.totalmem();
++		} catch (error) {}
++
++		try {
++			options.properties['common.shell'] = os.userInfo().shell;
++			options.properties['common.release'] = os.release();
++			options.properties['common.arch'] = os.arch();
++		} catch (error) {}
++
++		try {
++			const request = (/^http:/.test(this.endpoint) ? http : https).request(this.endpoint, {
++				method: 'POST',
++				headers: {
++					'Content-Type': 'application/json',
++				},
++			});
++			request.on('error', () => { /* We don't care. */ });
++			request.write(JSON.stringify(options));
++			request.end();
++		} catch (error) {}
++	}
++
++	public flush(options: { callback: (v: string) => void }): void {
++		if (options.callback) {
++			options.callback('');
++		}
++	}
++}
+Index: code-server/lib/vscode/src/vs/server/node/webClientServer.ts
+===================================================================
+--- code-server.orig/lib/vscode/src/vs/server/node/webClientServer.ts
++++ code-server/lib/vscode/src/vs/server/node/webClientServer.ts
+@@ -304,6 +304,7 @@ export class WebClientServer {
+ 					logoutEndpoint: this._environmentService.args['auth'] ? base + '/logout' : undefined,
+ 					proxyEndpointTemplate: base + '/proxy/{{port}}',
+ 					codeServerVersion: this._productService.codeServerVersion,
++					enableTelemetry: this._productService.enableTelemetry,
+ 					embedderIdentifier: 'server-distro',
+ 					serviceWorker: {
+ 						scope: vscodeBase + '/',
+Index: code-server/lib/vscode/src/vs/workbench/services/telemetry/browser/telemetryService.ts
+===================================================================
+--- code-server.orig/lib/vscode/src/vs/workbench/services/telemetry/browser/telemetryService.ts
++++ code-server/lib/vscode/src/vs/workbench/services/telemetry/browser/telemetryService.ts
+@@ -25,6 +25,7 @@ class WebAppInsightsAppender implements
+ 	private _telemetryCache: { eventName: string; data: any }[] = [];
+
+ 	constructor(private _eventPrefix: string, aiKey: string) {
++		console.log("halp")
+ 		const endpointUrl = 'https://vortex.data.microsoft.com/collect/v1';
+ 		import('@microsoft/applicationinsights-web').then(aiLibrary => {
+ 			this._aiClient = new aiLibrary.ApplicationInsights({
+@@ -74,6 +75,7 @@ class WebAppInsightsAppender implements
+ 		data = { ...data.properties, ...data.measurements };
+
+ 		// undefined assertion is ok since above two if statements cover both cases
++		console.log("are we here maybe?")
```

Comment:
nice, I usually write the same thing lol

Reply:
hahaha it would seem I forgot to refresh the patch after removing :rofl:

Reply:
Er I mean this is a feature :eyes: